src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/CatalogBackupTasklet.java

```diff
@@ -0,0 +1,651 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.tasklet;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.utils.BackupUtils;
+import org.geoserver.catalog.CoverageInfo;
+import org.geoserver.catalog.CoverageStoreInfo;
+import org.geoserver.catalog.DataStoreInfo;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.LayerGroupInfo;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.NamespaceInfo;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.config.GeoServerDataDirectory;
+import org.geoserver.config.GeoServerInfo;
+import org.geoserver.config.GeoServerPluginConfigurator;
+import org.geoserver.config.GeoServerPropertyConfigurer;
+import org.geoserver.config.LoggingInfo;
+import org.geoserver.config.ServiceInfo;
+import org.geoserver.config.SettingsInfo;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.geoserver.gwc.config.GWCInitializer;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Files;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.util.Filter;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.UnexpectedJobExecutionException;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.repeat.RepeatStatus;
+
+import com.thoughtworks.xstream.io.StreamException;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class CatalogBackupTasklet extends AbstractCatalogBackupRestoreTasklet {
```

Comment:
The name is confusing, this is actually a backup and restore tasklet (why not break them in two separate classes?) and it's not actually backing up the catalog, but global settings, logging, service definitions, gwc configurations and "additinal resources"